### ISSUE TITLE
Download update to ARK before installing update?

### ISSUE THREAD

**Issue by dahlskebank** [FezVrasta/ark-server-tools#237]
Is it possible to download the update for ARK without stopping the server? And then choose to stop server, install update and restart server? Connection to steam isn't always superb...

**Comment by klightspeed** [comment]
steamcmd does not provide a command to do this.

One way to implement this might be:
```
cd $steamcmdroot
./$steamcmdexec +login anonymous +force_install_dir $arkstagingdir +app_update $appid +quit

...

if [ ! -d "$arkstagingdir/steamapps/downloading/$appid" ]; then
  rsync -a "$arkstagingdir/." "$arkserverroot/."
fi
```

Another way that shouldn't take as much space, and relies on the way steamcmd updates (similar to how rsync updates by default), might be:
```
if [ ! -d "$arkstagingdir/ShooterGame" ]; then
  cp -al "$arkserverroot/ShooterGame/." "$arkstagingdir/ShooterGame"
  cp -al "$arkserverroot/Engine/." "$arkstagingdir/Engine"
  cp -al "$arkserverroot/linux64/." "$arkstagingdir/linux64"
  cp -al "$arkserverroot/PackageInfo.bin" "$arkstagingdir/PackageInfo.bin"
  cp -al "$arkserverroot/steamclient.so" "$arkstagingdir/steamclient.so"
  cp -a "$arkserverroot/steamapps/." "$arkstagingdir/steamapps"
  rm -rf "$arkstagingdir/ShooterGame/Content/Mods/"*
  rm -rf "$arkstagingdir/ShooterGame/Saved/"*
fi
cd $steamcmdroot
./$steamcmdexec +login anonymous +force_install_dir $arkstagingdir +app_update $appid +quit

...

if [ ! -d "$arkstagingdir/steamapps/downloading/$appid" ]; then
  cp -alu --remove-destination "$arkstagingdir/ShooterGame/." "$arkserverroot/ShooterGame"
  cp -alu --remove-destination "$arkstagingdir/Engine/." "$arkserverroot/Engine"
  cp -alu --remove-destination "$arkstagingdir/linux64/." "$arkserverroot/linux64"
  cp -alu --remove-destination "$arkstagingdir/PackageInfo.bin" "$arkserverroot/PackageInfo.bin"
  cp -alu --remove-destination "$arkstagingdir/steamclient.so" "$arkserverroot/steamclient.so"
  cp -au --remove-destination "$arkstagingdir/steamapps/." "$arkstagingdir/steamapps"  
fi
```

**Comment by Torath0071** [comment]
It's possible.

In my case i can't run my server with arkmanager start then i run it with command line
Then when server is started arkmanager can uppdate at same time. 
Next i need to kill process and restart when update is finished ^^

**Comment by klightspeed** [comment]
@Torath0071 That could cause undefined behaviour between Steam committing the update and you restarting the server.

**Comment by dahlskebank** [comment]
@klightspeed Thanks for finding a solution to this! I haven't tested it, will wait for it to be released in a proper version. Pretty sure players on our server would be unhappy if I fubared the server with some dev-version...

**Comment by Atriusftw** [comment]
We should focus on releasing a stable 1.4 before starting on this.

I have a feeling that this feature will take a lot of time to implement, if possible which will only delay the release of 1.4.

TL;DR: Milestone for 1.5 release.

**Comment by dahlskebank** [comment]
@Atriusftw: I agree. Not that I'm developing anything here. Auto-update of mods is way more appreciated! :)

**Comment by Atriusftw** [comment]
@dahlskebank Feedback is always welcome, and if you have experience in the relevant languages you're more then welcome to contribute by submitting a pull request to fix bugs, add a new feature or enhance existing features.
